### PR TITLE
Added MinMagnitude and MaxMagnitude ops

### PR DESCRIPTION
**Pull request by johnplatts**
Added the MinMagnitude and MaxMagnitude ops as the upcoming AVX10.2 instruction set will have VMINMAXPH/VMINMAXPS/VMINMAXPD instructions that can carry out the MinMagnitude and MaxMagnitude operations for F16/F32/F64 vectors.

### CODE REVIEW

File: hwy/ops/generic_ops-inl.h [hunk 1]
```diff
@@ -488,6 +488,63 @@ HWY_API V InterleaveEven(V a, V b) {
...
+#else
+#define HWY_NATIVE_FLOAT_MIN_MAX_MAGNITUDE
+#endif
+
+template <class V, HWY_IF_FLOAT_V(V)>
+HWY_API V MinMagnitude(V a, V b) {
+  const auto abs_a = Abs(a);
+  const auto abs_b = Abs(b);
+  return IfThenElse(Lt(abs_a, abs_b), a,
+                    Min(IfThenElse(Eq(abs_a, abs_b), a, b), b));
```
Comment (jan-wassenberg) [inline hwy/ops/generic_ops-inl.h:505]: Neat trick to match the x86 |a| < |b|, |b| < |a|, min(a, b) cases :)

**Review by jan-wassenberg — APPROVED** [review]
Thanks for adding!

**Comment by johnplatts** [comment]
@jan-wassenberg Are there any changes that need to be made before this pull request is ready to pull?

**Review by jan-wassenberg — APPROVED** [review]
Oops, I seem to have not yet set the correct label for import, sorry about that.

**Pull request merged**